dist/flow-spec/IBlock.d.ts

```diff
@@ -1,26 +1,26 @@
-import { IBlockExit, IBlockExitTestRequired, IContext, IContact } from '..';
-export interface IBlock {
+import { IBlockConfig, IBlockExit, IContact, IContext, ISetContactPropertyBlockConfig } from '..';
+export interface IBlock<BLOCK_CONFIG = IBlockConfig, BLOCK_EXIT_CONFIG = {}> {
```

Comment:
ALL_CAPS is an unusual casing for generic parameters, as it's usually used for constants.
But I understand that single letters (like `T`) are not clear enough.

And UpperCamelCase (`BlockConfig` and `BlockExitConfig`) can be confusing - looking like concrete implementations, i.e. class names.
On the other hand, we don't have these classes in this scope, so maybe we could use them (option 1)?

There is also `BlockConfigT` and `GBlockConfig` which I don't quite like. And `BlockConfigType` which is not too bad to my taste (option 2), what do you think?

Reply:
Otherwise loved the changes related to the generics!
It was nice to see a number of `any`, `unknown`, and `object` types removed

Reply:
The goal is to make it clear that these are not Class names, and they are not standard variables/fields. They are Generics, and especially because of the similar names colliding, it makes sense to make these stand out.